mesonbuild/mintro.py

```diff
@@ -146,11 +146,21 @@ def list_install_plan(installdata: backends.InstallData) -> T.Dict[str, T.Dict[s
             if key == 'headers':  # in the headers, install_path_name is the directory
                 install_path_name = os.path.join(install_path_name, os.path.basename(data.path))
 
-            plan[data_type] = plan.get(data_type, {})
-            plan[data_type][data.path] = {
+            entry = {
                 'destination': install_path_name,
                 'tag': data.tag or None,
             }
+
+            if key == 'install_subdirs' and data.exclude:
+                exclude_files, exclude_dirs = data.exclude
+                if exclude_dirs:
+                    entry['exclude_dirs'] = list(exclude_dirs)
+                if exclude_files:
+                    entry['exclude_files'] = list(exclude_files)
```

Comment:
So exclude_dirs / exclude_files may or may not exist, even for install_subdirs?

Reply:
This is the choice I made, I didn't see any other "optional" field whose implementation I could use as guidance. I can make it always present, if we like this better.

Reply:
Thinking about it, the `tag` field is resent eve when not specified. Maybe it is better to always have these fields for `install_subdir` entries. I updated the patch.

Reply:
Okay, thanks for investigating the different approaches and determining which one to select. :)